llvm/test/tools/lto/discard-value-names.ll

```diff
@@ -5,6 +5,8 @@
 ; RUN: %ld64 -lto_library %llvmshlibdir/libLTO.dylib -dylib -arch x86_64 -macos_version_min 10.10.0 -o %t.dylib %t.o -save-temps  -undefined dynamic_lookup -exported_symbol _bar -lSystem -mllvm -lto-discard-value-names=false
 ; RUN: llvm-dis %t.dylib.lto.opt.bc -o - | FileCheck --check-prefix=KEEP %s
 
+; REQUIRES: macos-sdk-10.15
```

Comment:
> Hmmm, did you try this with a newer SDK to confirm it works as expected or is possible that it now just doesn't run? ... 

@fhahn, you're right, testing locally this is too restrictive. The failing builder is using the MacOSX sdk version 13.3. Do you know if/how that version can have any effect on this test?

Reply:
I think the underlying issue should be fixed by 74f6ae9f245f2ae9e832ba297f451e04e008ada8. Please let me know if that's not case

Reply:
Thanks very much @fhahn , we'll re-enable the test on our builder and check the fix. I'll confirm in this thread.

Reply:
Confirmed the issue is fixed. Thanks @fhahn! Closing out this PR.